src/core/core.interaction.js

```diff
@@ -112,7 +109,7 @@ function getIntersectItems(chart, position, axis) {
 		}
 	};
 
-	const optimized = evaluateItemsAtIndex(chart, axis, position, evaluationFunc);
+	const optimized = binarySearchItems(chart, axis, position, evaluationFunc);
```

Comment:
I just realized this might not technically be correct (eventhough I introduced it). Imagine you have very large points that are all overlapping with the given position. Just returning the items at the nearest index might not be enough because there could be elements at several indices that are all intersecting with the given point

Reply:
Thats right, index and (plain) coordinate based determination can both be wrong.
I'll remove the intersect optimization for now. 
This case needs the generalized lookup, will do that in another pr.